### ISSUE TITLE
ALPHA TEST 01. Registration Process

### ISSUE THREAD

**Issue by thespacedoctor** [lsst-uk/lasair4#76]
Navigate to <https://lasair-dev.lsst.ac.uk>, try to create an account and sign in.

CHECKLIST:
- can you find the registration page easily?
- can you quickly and easily sign-up?
- can you login and out easily?
- any other feedback or improvements to the sign-up process you can suggest?

Note the registration page has a non-intrusive captcha (see bottom right corner) that can tell whether you are a human or a robot due to the way you interact with the page. Hence I have removed the need for email verification. This speeds up the sign-up process.

**Comment by gpfrancis** [comment]
- Registration appears to work OK (provided the user knows to click on "sign in", which is pretty common) with my default browser settings (Firefox)
- Login/logout all looks fine
- The sign in page appears to be trying and failing to download fonts from https://lasair-dev.lsst.ac.uk/lasair/static//webfonts/fa-solid-900.ttf and https://lasair-dev.lsst.ac.uk/lasair/static//webfonts/fa-solid-900.woff2 but this doesn't appear to actually cause any problems
- The registration page doesn't seem to handle short browser windows properly - the bottom of the page falls outside the broswer window with no scroll bar to access it
- I tried disabling scripts on the registration page to see whether it would cause the captcha to fail and it actually caused the username validation to fail (thinks I haven't entered a username when I have) which in turn causes the "sign up" button to fall off the bottom of the display (as in previous note).

**Comment by gpfrancis** [comment]
Just checked and the short broswer window/no scrollbar problem isn't Firefox specific, I got the same in Safari. Also just tried Edge on Windows and that has the same issue.

**Comment by mschwamb** [comment]
As reduced on mac chrome  as well

**Comment by thespacedoctor** [comment]
Great feedback already! Scrolling should be fixed if you would like to test.

**Comment by mschwamb** [comment]
The new scrolling works great.

**Comment by gpfrancis** [comment]
It's fixed on the registration page, but the sign in page still appears to have the same problem (although being shorter it only appears at smaller window sizes).

**Comment by genghisken** [comment]
Sign-up doesn't seem to work from a mobile. Captcha issues.
![Screenshot_20221216_194659_Chrome](https://user-images.githubusercontent.com/5502468/208177566-4a5e9bbc-2178-467e-b74d-f2fcd9079a1b.jpg)

**Comment by smarttgit** [comment]
All worked fine on iPad and Safari (20:20 on 16th Dec). No problem with capchta or scrolling. Very smooth, sign out and in again all ok. Can’t repeat any of the problems listed above

**Comment by mFulton07** [comment]
A few notes from me:

- The "Username" field prevents the entry of special characters, but the "First Name" and "Last Name" fields do not. While the fields appear to handle the input of special characters okay, we might want to consider restricting the fields to only letters and a few special characters (like hyphens), just like the Username.

- Entering something wrong on the form and trying to submit returns you to the form as expected, but there is an extra "This field is required" warning message above the sign-up button. I guess this is related to the hidden CAPTCHA field?

<img width="382" alt="Screenshot 2022-12-16 at 21 45 32" src="https://user-images.githubusercontent.com/58076215/208196471-3878f3b7-ecd4-4341-b42a-cfd911695d0d.png">

**Comment by RoyWilliams** [comment]
Rather unappetising front page on my phone. Need to click the top right triple-bar before the black menu comes up.
![Screenshot_20221217-093005_Samsung Internet](https://user-images.githubusercontent.com/8886481/208235900-af9b44c7-4e1a-4a6d-b894-d58dd7419cdf.jpg)

**Comment by thespacedoctor** [comment]
@gpfrancis  I have fixed the missing webfont issue. The console error should now be gone.
@gpfrancis  I have also fix scrolling on the login page (and a couple of other pages with the same issue).
@genghisken  We have all suspected that you are a robot for some time. I have reduced slightly the score required from the captcha to prove you are human. Hopefully you can signup now?

I have now added the 'Lasair Privacy Statement' that Roy was strong-armed into adding to the registration page. New users are required to confirm they have read the statement before being able to register. Navigate here and click on the 'Lasair Privacy Statement' link above the sign up bottom to read.

<https://lasair-dev.lsst.ac.uk/register/>

**Comment by mschwamb** [comment]
It would be good to get an acknowledgement email of registering so if someone did sign you up for an account and you didn't want one that you would be notified so one could take action

**Comment by thespacedoctor** [comment]
Thanks @mschwamb. @RoyWilliams requested email verification be added back into the workflow too. Unless others voice a strong opinion to the contrary I will readd it to the signup process.

**Comment by andyxerxes** [comment]
Howdy all. A bit late to the party... got overtaken by christmas and flu! Using Safari on Mac:

* Finding sign-in / sign-up page: at first it was hidden by Dave's techy sidebar, but I guess eventually that won't be an issue..
* 
* signing up: complained about short password; that's standard of course. Fixed that, but I got a HUGE long error page complaining of "integrity error". I can't figure out how to attach a file here, but I can email it if you want it. ... however, after ignoring this, I could log in with the name/pw I just created with no problem!
* 
* logging in / out did this multiple times, including after closing tab etc. 

Like others, I think email verification would be positively good. People expect it and its re-assuring that things are working.
